packages/webpack-cli/lib/groups/resolveOutput.js

```diff
@@ -0,0 +1,21 @@
+const path = require('path');
+
+/**
+ * Resolves the output flag
+ * @param {args} args - Parsed arguments passed to the CLI
+ */
+const resolveOutput = (args) => {
+    const { output } = args;
+    const finalOptions = {
+        options: { output: {} },
+        outputOptions: {},
+    };
+    if (output) {
+        const { dir, base, ext } = path.parse(output);
+        finalOptions.options.output.path = ext.length === 0 ? path.resolve(dir, base) : path.resolve(dir);
+        if (ext.length > 0) finalOptions.options.output.filename = base;
```

Comment:
Why we check on `ext`? I think `path.resolve` is enough for test

Reply:
if we pass absolute path name say `/user/code/dist` then `path.parse` will result in

`{dir: '/user/code', base: 'dist'}`

hence to support that we need this

Reply:
path.isAbsolute(output.path) ? output.path ? path.resolve(process.cwd(), 'output.path')

Reply:
Oh, do you mean `--output=/path/to/file.js`? If yes, it is invalid

Reply:
yep exactly 👍

Reply:
Output should be only directory, not filename

Reply:
I think we can support both, won't that be useful for users?

Reply:
It is invalid because we will specify HTML and CSS as entry points in the near future + `dependOn`, if you have HTML entrypoint and use `--output=/path/to/file.js` it was wrong, Ideally we should remove `--output` in favor `--output-path`, also this one will be consistent with webpack@5 cli args

Reply:
makes sense, will update 👍

Reply:
actually, we need to change a lot of tests because of this behaviour, let's do this after?

Reply:
Sounds fine, can you open new issue and focus on it?

Reply:
https://github.com/webpack/webpack-cli/issues/1844